pkgs/development/tools/build-managers/meson/fix-rpath.patch

```diff
@@ -0,0 +1,32 @@
+--- a/mesonbuild/compilers/compilers.py
++++ b/mesonbuild/compilers/compilers.py
+@@ -846,8 +848,10 @@
+             if paths != '':
+                 paths += ':'
+             paths += build_rpath
+-        if len(paths) < len(install_rpath):
+-            padding = 'X' * (len(install_rpath) - len(paths))
```

Comment:
Shouldn't hardcode `/nix/store` here, rather `substituteAll` the `$NIX_STORE` shell variable here.

Reply:
Do you mean `substituteAll`ing `builtins.storePath`, or reading `NIX_STORE` environment variable at runtime?

Reply:
I meant the value of `$NIX_STORE` during build-time, but the `builtins.storeDir` Nix variable seems to work as well.

Reply:
Done.